### ISSUE TITLE
Various minor issues

### ISSUE THREAD

**Issue by rs028** [AtChem/AtChem2#270]
- [ ] check that initial and and final model time are within the initial and final constraint timestamps before starting interpolation

**Comment by spco** [comment]
`Makefile.local` is solved by #285

**Comment by spco** [comment]
`constrainedFixedSpecies.config` is for constant-value species. Its name will become `speciesConstants.config` or `speciesFixed.config` in the Great Reorganisation.

**Comment by spco** [comment]
If you mean `mechanism-rate-declarations.f90`, then yes, it's used in `mechanism_rates()` in `solverFunctions.f90`.

**Comment by spco** [comment]
On 1., the current behaviour in `getConstrainedQuantAtT()` is this: if the required time is within the time covered by the constraint data, then it interpolates. Otherwise, it returns the value of the _last_ constraint data.

This certainly needs fixing in at least one case: if the required time is _before_ the first constraint data time, then the _last_ constraint data value is still returned. This is obviously not what is wanted.

I've opened #294 for this - see discussion there.

**Comment by spco** [comment]
On 8., both photolysis rates and environment variables are only evaluated when needed - each is interpolated individually, only when constrained. This happens each time `mechanism_rates()` is called from `FCVFUN()`, so the number of times that happens is controlled by CVODE as it completes the integration.

Species interpolation is very similar - it is called once for each of the constrained species in `FCVFUN()`, plus once when setting the initial conditions of each of the constrained species.

At this time I don't see a way to reduce the cost of this - once I've profiled more extensively, that will shine a light on how expensive this all is, and whether it's worth looking harder at.

**Comment by rs028** [comment]
@spco I added this to the Milestone for version 1.1 because there are only a few points left and some of them are just questions to improve the documentation. If some require more time than you have at the moment, they can be moved to their own issue.

**Comment by spco** [comment]
On number 11 above - it shouldn't matter if the same species is mentioned twice in the same file of either `speciesConstant.config`, `speciesConstrained.config`, `photolysisConstrained.config`. It's inefficient, as it'll all be read in and stored and processed, but it shouldn't matter to the result. But it  looks like we don't check that there's no conflict between species defined as constant or variably constrained? I'll make this a separate issue.

**Comment by spco** [comment]
On 3. above, currently we process RH, and then H2O. RH can be supplied as a constrained, or fixed value. H2O is then either calculated from RH (via `convertRHtoH2O()`), or set as constrained, set as fixed, or given a default value. There are 2 issues with this setup currently:
1) there's nothing to stop the user setting both to fixed or constrained values that are incompatible.
2) it's a one-way relationship, because H2O can never be used to calculate RH.

The setting of environment variables is one of the parts of this code that is a bit wonky in general. Ideally it wants a bit of an overhaul and rethinking to ensure that the user has decent flexibility in what they can supply as inputs, but cannot do things which are incompatible. We also already have a future plan to allow users to possibly add new environment variables themselves, so I'd suggest that we bundle all of these up into one work package in v2, rather than try to do much here. What do you think @rs028?

**Comment by spco** [comment]
On 6. above, we have the following naming:
```
environmentVariables.config: 'PRESS', 'TEMP', 'RH', 'BLHEIGHT', 'DILUTE', 'ROOFOPEN', 'M', 'H2O', 'DEC', 'JFAC'
mechanism_rates:             pressure, temp,   RH,   BLH,        DILUTE,   ROOFOPEN,   M,   H2O,   DEC,   JFAC
```
so the differences are in `pressure` and `blheight`. Looking to a future where we might be able to implement new enviroment variables, the only sensible option is that the config and the code have to talk in the same language, otherwise new environment variables would need a shorthand name for the code or something like that - messy.
To maintain compatibility with existing config files, I'd suggest that we always use 'PRESS' (or lowercase) and 'BLHEIGHT' (or lowercase). There are only a few places where this needs changing, so I'll raise a PR for that. Let me know if you have any objection.

**Comment by spco** [comment]
Re the above on point 6, current the python `mech_converter.py` uses `BLH` as a reserved keyword, but I think it's fine to just replace this with `BLHEIGHT`.

**Comment by spco** [comment]
Re 13. above, I'd keep this for v2, but ideally this should be possible to make the scripts version-independent, as they don't have too many dependencies. If it's not possible, I'd recommend upgrading to python 3 in AtChem2v2.

**Comment by rs028** [comment]
Okay, so #354 solves 6 and 13 can be made into a separate (low priority issue).
That leaves 3 and 5. Do you want to do those now or leave for future versions?

**Comment by spco** [comment]
I would suggest we leave those both for the future - both are embroiled in the environment variables section, and should be treated together when we have a plan for that.

**Comment by spco** [comment]
And as such, I think our v1.1 milestones have all been completed?

**Comment by rs028** [comment]
yes, I think we are good to go for v1.1. Unless there is something else you have time to do this week.

**Comment by spco** [comment]
I think my time here is finished for a while, so it's as good a time as any to release v1.1 from my perspective!
